drivers/dp/swdp_bitbang.c

```diff
@@ -562,113 +562,120 @@ static int sw_configure(const struct device *dev,
 static int sw_port_on(const struct device *dev)
 {
 	const struct sw_config *config = dev->config;
-
-	gpio_pin_set_dt(&config->clk, 1);
+	int ret;
 
 	if (config->dnoe.port) {
-		gpio_pin_set_dt(&config->dnoe, 1);
+		ret = gpio_pin_configure_dt(&config->dnoe, GPIO_OUTPUT_ACTIVE);
+		if (ret) {
+			return ret;
+		}
 	}
-
 	if (config->dout.port) {
-		gpio_pin_set_dt(&config->dout, 1);
-	} else {
-		int ret;
-
-		ret = gpio_pin_configure_dt(&config->dio, GPIO_OUTPUT_ACTIVE);
+		ret = gpio_pin_configure_dt(&config->dout, GPIO_OUTPUT_ACTIVE);
 		if (ret) {
 			return ret;
 		}
 	}
-
-	if (config->noe.port) {
-		gpio_pin_set_dt(&config->noe, 1);
-	}
-	if (config->reset.port) {
-		gpio_pin_set_dt(&config->reset, 1);
-	}
-
-	return 0;
-}
-
-static int sw_port_off(const struct device *dev)
-{
-	const struct sw_config *config = dev->config;
-
-	if (config->dnoe.port) {
-		gpio_pin_set_dt(&config->dnoe, 0);
+	ret = gpio_pin_configure_dt(&config->dio, GPIO_INPUT);
+	if (ret) {
+		return ret;
 	}
-
-	if (config->dout.port) {
-		gpio_pin_set_dt(&config->dout, 0);
-	} else {
-		int ret;
-
-		ret = gpio_pin_configure_dt(&config->dio, GPIO_INPUT);
+	if (config->noe.port) {
+		ret = gpio_pin_configure_dt(&config->noe, GPIO_OUTPUT_ACTIVE);
 		if (ret) {
 			return ret;
 		}
 	}
-
-	if (config->noe.port) {
-		gpio_pin_set_dt(&config->noe, 0);
+	ret = gpio_pin_configure_dt(&config->clk, GPIO_OUTPUT_ACTIVE);
+	if (ret) {
+		return ret;
 	}
-	if (config->reset.port) {
-		gpio_pin_set_dt(&config->reset, 1);
+	ret = gpio_pin_configure_dt(&config->reset, GPIO_OUTPUT_ACTIVE);
+	if (ret) {
+		return ret;
 	}
 
 	return 0;
 }
 
-static int sw_gpio_init(const struct device *dev)
+static int sw_port_off(const struct device *dev)
 {
 	const struct sw_config *config = dev->config;
-	struct sw_cfg_data *sw_data = dev->data;
 	int ret;
 
-	ret = gpio_pin_configure_dt(&config->clk, GPIO_OUTPUT_ACTIVE);
-	if (ret) {
-		return ret;
-	}
-
-	ret = gpio_pin_configure_dt(&config->dio, GPIO_INPUT);
-	if (ret) {
-		return ret;
-	}
-
-	if (config->dout.port) {
-		ret = gpio_pin_configure_dt(&config->dout, GPIO_OUTPUT_ACTIVE);
+	/* If there is a transceiver connected to IO, pins should always be driven. */
+	if (config->dnoe.port) {
+		ret = gpio_pin_configure_dt(&config->dnoe, GPIO_OUTPUT_INACTIVE);
 		if (ret) {
 			return ret;
 		}
-	}
-
-	if (config->dnoe.port) {
-		ret = gpio_pin_configure_dt(&config->dnoe, GPIO_OUTPUT_INACTIVE);
+		if (config->dout.port) {
```

Comment:
```c
	if (config->dnoe.port) {
		ret = gpio_pin_configure_dt(&config->dnoe, GPIO_OUTPUT_INACTIVE);
		if (ret) {
			return ret;
		}

		if (config->dout.port) {
```
Please fix in all your changes accordingly.

Reply:
I can't spot the difference. What should I change?

Reply:
Okay:

```c
		if (ret) {
			return ret;
		}

		if (foo->bar) {
			/* */
		}
	}

	return -42
```

Not okay:

```c
		if (ret) {
			return ret;
		}
		if (foo->bar) {
			/* */
		}
	}
	return -42
```

Reply:
Ah, you would like some space between the blocks, gotcha.

Reply:
Still not addressed, please take a look at the changes in sw_port_on().

Reply:
it it okay now? @jfischer-no